src/loki/LokiBase.cpp

```diff
@@ -175,38 +211,115 @@ void LokiBase::processingThread() {
   TSCTimer ProduceTimer;
   while (runThreads) {
     if (InputFifo.pop(DataIndex)) { // There is data in the FIFO - do processing
-      auto DataLen = EthernetRingbuffer.getDataLength(DataIndex);
+      auto DataLen = RxRingbuffer.getDataLength(DataIndex);
       if (DataLen == 0) {
         Counters.FifoSeqErrors++;
         continue;
       }
 
-      /// \todo use the Buffer<T> class here and in parser
-      auto __attribute__((unused)) DataPtr = EthernetRingbuffer.getDataBuffer(DataIndex);
-      /// \todo add parser
+      /// \todo use the Buffer<T> class here and in parser?
+      /// \todo avoid copying by passing reference to stats like for gdgem?
+      auto DataPtr = RxRingbuffer.getDataBuffer(DataIndex);
+      auto Res = ESSReadout.validate(DataPtr, DataLen, ReadoutParser::Loki4Amp);
```

Comment:
I do not really like the name ESSReadout for the parser instance. For me "readout" implies read out data. I had to look into LokiBase.h to see that is actually the parser. Could we not call it 

ReadoutParser ESSReadoutParser;

That's why I do not like UpperCamelCase for variables and class names, otherwiser the obvious choice would be

ReadoutParser readoutParser;

OK, enough whining..

Reply:
Yes, I can rename this. We could also have a discussion on how to name things
more consistently in the future.